### PR TITLE
Update documentation

### LINKED ISSUE #494: A few random documentation fixes
Appended as git patches against v1.2.0-234-g5167f6c (develop branch).
[0001-Documentation-Fix-an-exponent.patch.txt](https://github.com/libtom/libtommath/files/5719805/0001-Documentation-Fix-an-exponent.patch.txt)
[0002-Documentation-Fix-dev-urandom.patch.txt](https://github.com/libtom/libtommath/files/5719807/0002-Documentation-Fix-dev-urandom.patch.txt)
[0003-Documentation-Fix-Format-mp_rand_source.patch.txt](https://github.com/libtom/libtommath/files/5719808/0003-Documentation-Fix-Format-mp_rand_source.patch.txt)
[0004-Documentation-Grammo.patch.txt](https://github.com/libtom/libtommath/files/5719810/0004-Documentation-Grammo.patch.txt)
[0005-Documentation-Kludge-for-decent-mod-operation.patch.txt](https://github.com/libtom/libtommath/files/5719811/0005-Documentation-Kludge-for-decent-mod-operation.patch.txt)

### PR DESCRIPTION
**Pull request by sjaeckel**
PR with patches from #494 

This closes #494

### CODE REVIEW

**Review by sjaeckel — COMMENTED** [review]
LGTM

File: doc/bn.tex
Comment (czurnieden) [inline doc/bn.tex:2607]: Why not `\mod`? Ah, I see: `\def\mod{{\mathit\ mod\ }}` that's not the most elegant solution. It is used as the symbol for "remainder" so I would propose `\def\mod#1{\allowbreak\mkern18mu{\rm mod}\,\,#1}` instead, maybe with a bit less kerning.

```
\documentclass{article}
% The definition of "\pmod" from plain TeX without the parentheses.
\def\mod#1{\allowbreak\mkern18mu{\rm mod}\,\,#1}
\def\tightmod#1{\allowbreak\mkern12mu{\rm mod}\,\,#1}
\begin{document}
With \texttt{\textbackslash mod}.
$$ a = 0 \mod{2}$$
With \texttt{\textbackslash tightmod} which has a third less
space before ``mod''.
$$ a = 0 \tightmod{2}$$
No extra packages used.
\end{document}
```
There is also a curious redefinition of `\pmod` which is nowhere to be seen in use. It is, well should be, a `\mod` with parentheses to denote an  equivalence class. I haven't taken a deeper look into ``bn.tex` for some time but it (the original with parentheses) could be useful in some places.

**Review by czurnieden — APPROVED** [review]
I approve this time but `bn.tex` is a mishmash of different hands with different styles and really in need of a thorough overhaul.
Any volunteers?

File: doc/bn.tex
Comment (sjaeckel) [reply doc/bn.tex:2607]: I'm going to merge this and after your latest is merged I'll have a look over the file. I already started changing stuff

**Pull request merged**
